js/src/jit/BaselineCacheIRCompiler.cpp

```diff
@@ -2499,54 +2503,67 @@ static bool AddToFoldedStub(JSContext* cx, const CacheIRWriter& writer,
   return true;
 }
 
-ICAttachResult js::jit::AttachBaselineCacheIRStub(
-    JSContext* cx, const CacheIRWriter& writer, CacheKind kind,
-    JSScript* outerScript, ICScript* icScript, ICFallbackStub* stub,
-    const char* name) {
-  // We shouldn't GC or report OOM (or any other exception) here.
-  AutoAssertNoPendingException aanpe(cx);
-  JS::AutoCheckCannotGC nogc;
+#ifdef ENABLE_JS_AOT_ICS
+void DumpNonAOTICStubAndQuit(CacheKind kind, const CacheIRWriter& writer) {
+  // Generate a filename based on the current time and PID.
+  char filename[64];
+  snprintf(filename, sizeof(filename), "IC-%" PRIu64,
+           mozilla::RandomUint64OrDie());
+  FILE* f = fopen(filename, "w");
+  MOZ_RELEASE_ASSERT(f);
+ 
+  // Generate the CacheIR text to dump to a file.
+  {
+    Fprinter printer(f);
+    SpewCacheIROpsAsAOT(printer, kind, writer);
+  }
+  fflush(f);
+  fclose(f);
+  fprintf(stderr, "UNEXPECTED NEW IC BODY\n");
 
-  if (writer.tooLarge()) {
-    return ICAttachResult::TooLarge;
+  fprintf(stderr,
+          "Please add the file '%s' to the ahead-of-time known IC bodies in "
+          "js/src/ics/.\n",
+          filename);
+
+  if (!getenv("AOT_ICS_KEEP_GOING")) {
```

Comment:
Is there a place we should/could document this environment variable?

Reply:
Good idea; added a note to the error message just above.